cmd/thanos/bucket.go

```diff
@@ -234,15 +235,18 @@ func registerBucketLs(m map[string]setupFunc, root *kingpin.CmdClause, name stri
 			}
 		}
 
-		if err := bkt.Iter(ctx, "", func(name string) error {
-			id, ok := block.IsBlockDir(name)
-			if !ok {
-				return nil
-			}
+		metas, _, metaFetchErr := fetcher.Fetch(ctx)
+		// For partial view allow adding new blocks at least.
+		if metaFetchErr != nil && metas == nil {
+			return metaFetchErr
+		}
+
+		for id, meta := range metas {
 			objects++
-			return printBlock(id)
-		}); err != nil {
-			return errors.Wrap(err, "iter")
+			err := printBlock(id, meta)
```

Comment:
Super small nit: if you have just single return argument and it is an error there is nice pattern that simplifies this:

```
if err := printBlock(...); err != nil {
   return errors.Wrap(err, "print")
} 
```